external.go

```diff
@@ -28,7 +30,9 @@ type CommandExecutor interface {
 // EventRecorder is an interface implemented by the engine and used by the
 // application to record events outside of a Dogma message handler.
 //
-// It is RECOMMENDED that engines provide an implementation of this interface.
+// Engine implementations MAY provide mechanisms for recording events outside of
```

Comment:
In projections events are called **event messages**. Should we try to call them consistently across dogma library?

Reply:
Yep, for sure! I **have** been making a distinction between when we're talking about the command or event as a concept, versus the message itself. 

For example, in the `AggregateCommandScope.RecordEvent()` method, I've used the language "record .. a [domain] event", without the word "message" - as it's the conceptual **event** that we want to record. It "just so happens" to be represented by a message.

```go
// RecordEvent records the occurrence of a domain event as a result of the
// domain command message that is being handled.
```
(We also have the word "domain" in the mix here, I'm not sure if that should be retained).

The same can be said for "executing a command".  It's not the "message" that we want to execute, but the command it represents.

The difference is subtle, and possibly not worthwhile. What's more, I'm not sure that I've been consistent in these distinctions. Suggestions welcome :)